lib/storage/metadata/file/MetadataFileServer.js

```diff
@@ -187,9 +198,24 @@ class MetadataFileServer {
             },
             batch: (request, logger, callback) => {
                 const dbPath = request.db.split(SUBLEVEL_SEP);
-                const subDb = dbService.lookupSubLevel(dbPath);
-                subDb.batch(request.array, SYNC_OPTIONS,
-                            err => callback(err));
+                const ops = request.array.map(
+                    op => Object.assign({}, op, { prefix: dbPath }));
+                if (this.recordLog.enabled) {
+                    this.recordLogService
+                        .withRequestParams(
+                            { logName: this.recordLog.recordLogName })
+                        .createLogRecordOps(
+                            ops, (err, logEntries) => {
+                                debug('ops to batch:', ops);
```

Comment:
The `env` param is special, it is present only on the server-side backend call to pass some environment attributes. It gets passed as the first argument of every backend API function. The client does not see this parameter and does not pass it in the call.